File: challenges/rate-limiter/go/starter/exercise.go
package exercise

import "refresher/tooling/go/contracts"

// Add the state required by the challenge contract.
type SlidingWindowLimiter struct{}

func NewSlidingWindowLimiter(limit int, windowMS int64, clock contracts.Clock) *SlidingWindowLimiter {
	panic("Not implemented")
}

func (l *SlidingWindowLimiter) Allow(client string) bool {
	panic("Not implemented")
}

File: challenges/rate-limiter/php/starter/exercise.php
<?php
declare(strict_types=1);

namespace Refresher;

use Closure;
use LogicException;

final class SlidingWindowLimiter
{
    public function __construct(
        int $limit,
        int $windowMs,
        Closure $clock,
    ) {
        throw new LogicException("Not implemented");
    }

    public function allow(string $client): bool
    {
        throw new LogicException("Not implemented");
    }
}
